From 502d1432d1fac1876296c7196ece091e5504a1b4 Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Wed, 24 Oct 2018 11:12:20 +0530
Subject: [PATCH 0481/1852] drm: xlnx: dsi: Disable bridge instance

commit 85f58df6620cc4819f1ecbd49de912afb2b7817f from
https://github.com/Xilinx/linux-xlnx.git

This patch disables the bridge instance when
dsi controller is disabled.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_dsi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_dsi.c b/drivers/gpu/drm/xlnx/xlnx_dsi.c
index ab758b3f332c..4edd5d27d6b5 100644
--- a/drivers/gpu/drm/xlnx/xlnx_dsi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_dsi.c
@@ -653,6 +653,9 @@ static void xlnx_dsi_disable(struct drm_encoder *encoder)
 {
 	struct xlnx_dsi *dsi = encoder_to_dsi(encoder);
 
+	if (dsi->bridge)
+		xlnx_bridge_disable(dsi->bridge);
+
 	xlnx_dsi_set_display_disable(dsi);
 }
 
-- 
2.31.1

